cfme/tests/automate/test_service_automate.py

```diff
@@ -455,3 +459,185 @@ def test_retire_vm_now(setup_provider, full_template_vm, new_user):
+                              secret=fauxfactory.gen_alphanumeric(start="pwd")),
+        email=fauxfactory.gen_email(),
+        groups=new_group_tenant,
+        cost_center="Workload",
+        value_assign="Database",
+    )
+    yield user
+    user.delete_if_exists()
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1492158])
+@pytest.mark.parametrize('context', [ViaSSUI, ViaUI])
+@pytest.mark.parametrize("file_name", ["bz_1492158.yml"], ids=["dialog"])
+@pytest.mark.parametrize(
+    ['set_child_tenant_quota', 'custom_prov_data'],
+    [
+        [(('cpu', '2'), ('storage', '0.01'), ('memory', '2')),
+         ({'hardware': {'num_sockets': '8', 'memory': '4096'}})]
+    ],
+    indirect=['set_child_tenant_quota', 'custom_prov_data'],
+    ids=['max_cpu_storage_memory']
+)
+@pytest.mark.provider([RHEVMProvider], scope="module")
+def test_quota_calculation_using_service_dialog_overrides(
+        request, appliance, setup_provider, provider, domain_setup, set_child_tenant_quota, context,
+        custom_prov_data, import_dialog, file_name, catalog, new_child_tenant_user):
+    """
+    This test case is to check Quota calculation using service dialog overrides.
+    Bugzilla:
+        1492158
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/6h
+        caseimportance: medium
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.8
+        casecomponent: Infra
+        tags: quota
+        testSteps:
+            1. create a new domain quota_test
+            2. Using the Automate Explorer, copy the
+               ManageIQ/System/CommonMethods/QuotaMethods/requested method
+               to the quota_test domain.
+            3. Import the attached dialog. create catalog and catalog
+               item using this dialog
+            4. create a child tenant and set quota. create new group and
+               user for this tenant.
+            5. login with this user and provision by overriding values
+            6. Also test the same for user and group quota source type
+        expectedResults:
+            1.
+            2.
+            3.
+            4.
+            5. Quota should be denied with reason for quota exceeded message
+            6. Quota should be denied with reason for quota exceeded message
+    """
+    sd, ele_label = import_dialog
+    prov_data = custom_prov_data
+    catalog_item = appliance.collections.catalog_items.create(
+        provider.catalog_item_type,
+        name=fauxfactory.gen_alphanumeric(start="test_"),
+        description=fauxfactory.gen_alphanumeric(start="desc_"),
+        display_in=True,
+        catalog=catalog,
+        dialog=sd,
+        prov_data=prov_data
+    )
+    request.addfinalizer(catalog_item.delete_if_exists)
+    with new_child_tenant_user:
+        with appliance.context.use(context):
+            appliance.server.login(new_child_tenant_user)
```

Comment:
Why do you login again when you're already using `with new_child_tenant_user`? I think using that context manager should be enough, it will log you in.

Reply:
It is necessary to login with SSIU.

Reply:
What behavior are you seeing, because yes, the context managers should control login.

I think you need to use the context first though, and the user context within it.